### ISSUE TITLE
loading third party extensions

### ISSUE THREAD

**Issue by pkra** [mathjax/MathJax-node#134]
Just a note to think about how users can specify third party extensions (i.e., without copying extensions into the right folder of core MathJax).

**Comment by pkra** [comment]
I'm thinking a simple configuration option that can be used to specify a folder (where developers store their third party extensions) should work ok. That folder can be resolved and set as `Contrib` path in core MathJax.

I'm not sure how jsdom behaves in terms of security (since mathjax-node works on `fs://`).

**Comment by dpvc** [comment]
You may not want to override `[Contrib]` but rather make a new path for these.  But yes, a way to do that would be useful.

**Comment by pkra** [comment]
> You may not want to override `[Contrib]` but rather make a new path for these

My reasoning was 

a) the hard-coded `[Contrib]` path for the CDN's contrib folder shouldn't work in mathjax-node -- I had actually assumed we disallow remote sources when opening the jsdom window (for security reasons) but I see now that's not the case; I'll create an issue.
b) re-using `[Contrib]` makes it easy to use extension from the third party extensions repo.

**Comment by dpvc** [comment]
>  the hard-coded `[Contrib]` path for the CDN's contrib folder shouldn't work in mathjax-node

I'm not sure why not.

> re-using `[Contrib]` makes it easy to use extension from the third party extensions repo.

Using another path would not make it any harder.  You will have to give an explicit path to the extension anyway, so I'm not sure why `[OtherPath]/extension.js` is harder than `[Contrib]/extensino.js`.  The only extensions that actually use `[Contrib]` internally are the a11y extensions, and they use `[a11y]` for their paths, so configuring that would make them work even if they are loaded from `[OtherPath]`.  So I don't see what is gained, really.  Not overriding `[Contrib]` allows extensions to be loaded directory from the CDN; overriding it would force them to be installed locally (more work and chance of error).

**Comment by pkra** [comment]
> Using another path would not make it any harder. Y

Maybe I still don't understand the internals then. 

All third party extensions use `MathJax.Ajax.loadComplete("[Contrib]/...");`. Wouldn't a mathjax-node user have to change that?

**Comment by dpvc** [comment]
Ah yes, you are correct.  I had forgotten about that.  `[Contrib]` it is, then.

**Comment by pkra** [comment]
Moving this out of the milestone.

**Comment by pkra** [comment]
Since the CDN was retired, third party extensions should now have their own prefix so this should be easier.

**Comment by pkra** [comment]
I've made a PR for a first idea for htis.

The main idea was to make it easier to use an extension via NPM, i.e., install it as usual and then do something like

```
   const path = require('path);
   const mj = require("mathjax-node");

    mj.config( {
        extensions: '[test]/test.js',
         paths: {
            'test': path.dirname(require.resolve('test'))
        }
    });
```

Not elegant but perhaps good enough.

**Comment by dpvc** [comment]
Resolves by PR #391
